well_known.go

```diff
@@ -28,27 +28,27 @@ func LookupWellKnown(serverNameType ServerName) (*WellKnownResult, error) {
 	// Request server's well-known record
 	resp, err := http.Get(wellKnown)
 	if err != nil {
-		return nil, err
+		return nil, false, err
```

Comment:
why does this return an err, but a non-200 response does not? they both amount to the same thing: that the well-known request was unsuccessful.

The problem is not that LookupWellKnown returns an error when there is no well-known. The problem is that the federation tester takes the error and sticks it straight into a field called 'error' in the json.